builtin/process_osh.py

```diff
@@ -682,4 +694,109 @@ def Run(self, cmd_val):
         return 0
 
 
+class Kill(vm._Builtin):
+    """Send a signal to a process"""
+
+    def __init__(self, job_list):
+        # type: (process.JobList) -> None
+        self.job_list = job_list
+
+    def _SignameToSignum(self, name):
+        # type: (str) -> int
+        signal_name = name.upper()
+        if signal_name.startswith("SIG"):
+            signal_name = signal_name[3:]
+        return signal_def.GetNumber(signal_name)
+
+    def _ParsePid(self, pid_arg, pid_arg_loc):
+        # type: (str, loc_t) -> int
+        if pid_arg.startswith("%"):
+            job = self.job_list.JobFromSpec(pid_arg)
+            if job is None:
+                e_usage("invalid signal specification %r" % pid_arg,
+                        pid_arg_loc)
+            else:
+                return job.ProcessGroupId()
+        else:
+            try:
+                target_pid = int(pid_arg)
+                return target_pid
+            except ValueError:
+                e_usage("invalid process id specification %r" % pid_arg,
+                        pid_arg_loc)
+
+    # sigspec can either be in the form 15, TERM, or SIGTERM (case insensitive)
+    # returns signal_def.NO_SIGNAL if sigspec is in invalid format
+    def _SigspecToSignal(self, sigspec):
+        # type: (str) -> int
+        signal = signal_def.NO_SIGNAL
+        if sigspec.isdigit():
+            signal = int(sigspec)
+        else:
+            signal = self._SignameToSignum(sigspec)
+        return signal
+
+    def Run(self, cmd_val):
+        # type: (cmd_value.Argv) -> int
+        signal_to_send = 15  # sigterm, the default signal to send
+        arg_r = args.Reader(cmd_val.argv, locs=cmd_val.arg_locs)
+        arg_r.Next()  # skip command name
+        first_positional, first_positional_loc = arg_r.ReadRequired2(
+            "you must provide a process id")
```

Comment:
This error looks like this:

```
$ bin/osh -c 'kill -9'
  kill -9
  ^~~~
[ -c flag ]:1: 'kill' you must provide a process id
```

It is meant to enforce a certain grammatical convention

If you write `ReadRequired('expected a process ID')` then it will read grammatically